keel-bakery-plugin/src/main/kotlin/com/netflix/spinnaker/keel/bakery/api/ImageSpec.kt

```diff
@@ -1,9 +1,12 @@
 package com.netflix.spinnaker.keel.bakery.api
 
+import com.netflix.spinnaker.keel.api.Applicationed
+
 data class ImageSpec(
   val artifactName: String,
   val baseLabel: BaseLabel,
   val baseOs: String,
   val regions: Set<String>,
-  val storeType: StoreType
-)
+  val storeType: StoreType,
+  override val application: String // the application an image is baked in
```

Comment:
this might be tricky - there are definitely cases of generic-ish images that are used in multiple applications (and maybe even baked in multiple different applications pipelines). Not sure what to suggest here though..

Reply:
Would that be an issue? Pretty sure the application field is irrelevant as far as the bakery is concerned, it just determines where the bake task show up in the Spinnaker UI.

Reply:
discussed this in person with @emjburns and I think I get where we are at with this now (this is where we will ask to bake a newly detected artifact vs where we expect it to have come from)

Reply:
Yep, adding this so that we bake artifacts where people want them. This will be used for generating bake tasks. I should actually do that...